app/classes/Comment/model.js

```diff
@@ -0,0 +1,75 @@
+const validator = require('validator')
+const mongoose = require('mongoose')
+const Schema = mongoose.Schema
+
+// class Model - Abstract Class - cannot be instantiated
+//   public - accessible to inheriting classes
+//     getModel()
+//   private
+//     schema
+//     model
+
+class Model {
+  constructor () {
+    if (this.constructor === Model) {
+      throw new Error("Can't instantiate abstract class!")
+    }
+  }
+
+  #schema = new Schema({
```

Comment:
I think the model and schema is not private because this syntax for private is only applicable in arrow functions.

Reply:
Check the instance of Run is able to access #model and #schema 
Sample code
![image](https://user-images.githubusercontent.com/31209617/89094825-bec87080-d3e5-11ea-825c-9f56677a0023.png)

Reply:
Could not reproduce this, It results in an SyntaxError for me, tested in fresh environments as well
![Screenshot from 2020-08-01 13-06-09](https://user-images.githubusercontent.com/39027928/89096955-343d3c80-d3f8-11ea-9fd9-563539d589ea.png)

Reply:
yes, I think this private syntax applies to class field variables as well and are supported in node v12+ checkout out these links 
https://github.com/tc39/proposal-class-fields#private-fields
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields#Private_instance_fields

Reply:
@ksraj123 What we could do is that schema and model should not be inherited as of now it's getting inherited also it could be overwritten.

Reply:
I am not sure if that can be done. I think that if one class extends another one then it inherits all of its properties but the private properties are not directly accessible, we have the getter `getModel` for that and private properties could not be overwritten directly. I tested that part with the actual code as well and got errors. Though I am not sure how that is working for you. I think our Controller has to inherit our Model as we do not want the Model to be instantiable so we have to make it abstract and I think public members of an abstract class could only be accessed if another class inherits it. Model, Controller and Router should have meaning only to the Comment and should not be accessible directly.